src/helpers/check_diff.jl

```diff
@@ -0,0 +1,136 @@
+
+# Fits \hat{y} = a + b x.
+# Returns a, b, and the R^2 of the fit.
+function solve_simple_lm(x, y)
+    # See https://en.wikipedia.org/wiki/Simple_linear_regression
+    correlation = cor(x, y)
+    beta = std(y) / std(x) * correlation
+    alpha = mean(y) - beta * mean(x)
+    r2 = size(x, 1) <= 2 ? 1 : correlation^2
+    return (alpha, beta, r2)
+end
+
+function linearize_function(x, y; r2_cutoff=0.99)
+    @assert size(x) == size(y)
+    y = y[sortperm(x)]
+    x = sort(x)
+    seq_len = size(y, 1)
+    n_elem = (seq_len - 1) * (seq_len) ÷ 2
+    df = DataFrame(;
+        a=zeros(n_elem),
+        b=zeros(n_elem),
+        r2=zeros(n_elem),
+        left=zeros(n_elem),
+        right=zeros(n_elem),
+        length=zeros(n_elem),
+        left_idx=zeros(Int32, n_elem),
+        right_idx=zeros(Int32, n_elem),
+    )
+    counter = 1
+    for i in 1:seq_len
```

Comment:
...this seems to build _all_ possible lines through all data? For two points your initial function to finde the regression line is a little too much (2 points uniquely define a line).